### PR TITLE
CORTX-23443:  adding retry decorator in consul confstore in main

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

# Problem Statement
- Since confstore branch is discarded, raised a separate PR for retry mechanism in consul.

# Design
-  These changes are pre approved changes from confstore branch, now raised the same changes against main branch.
relevant PRs: https://github.com/Seagate/cortx-utils/pull/776, https://github.com/Seagate/cortx-utils/pull/753

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing
- Are test cases updated along with code changes due to Enhancements/Bugs [Y/N]:
- Confirm that new test cases are added to regression and sanity plan files and relevant feature plan files [Y/N]:
- Confirm that Test Cases are added for new features added [Y/N]: 
- Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- Confirm Testing was performed with installed RPM/K8s deployment [Y/N]:  

# Review Checklist 
####  Before posting the PR please ensure:
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] New package/s added to setup.py?
- [x] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified.

# KVstore/Confstore feature (changes/fixes) checklist
#### Confirm changes are made for:
- [ ] ConfStore
- [ ] KVStore
- [ ] ConfCli
- [ ] Test cases added for all above 3
- [ ] changes done in all the KVpayloads (ConsulKvPayload, IniKvPayload, KvPayload)

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Comment by rahul-tripathi-git** [comment]
Images build job: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5812
Deployment job: https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/2666

**Comment by rahul-tripathi-git** [comment]
These changes are pre approved changes from confstore branch, now raised the same changes against main branch.
relevant PRs: https://github.com/Seagate/cortx-utils/pull/776, https://github.com/Seagate/cortx-utils/pull/753

**Comment by rahul-tripathi-git** [comment]
Tried manual deployment as well, retry is working as expected.

**Review by rohit-k-dwivedi — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]

**Review by suryakumar1024 — APPROVED** [review]
LGTM

File: py-utils/src/utils/common/common.py [hunk 1]
```diff
@@ -1,9 +1,53 @@
...
+        ...
+    or
+    @ExponentialBackoff(Exception, tries=2, delay=10, backoff=2, cap=2)
+    def bar():
+        ...
+    Args:
+        exception (Exception or tuple(Exception)):  it can be a single exception
+                                                    or a tuple of exceptions.
+        tries (int): 10     number of times to try before failing.
+        delay (int): 1      initiall delay (in seconds) between retries.
```
Comment (cdeshmukh) [inline py-utils/src/utils/common/common.py:26]: typo initiall -> initial

File: py-utils/src/utils/common/common.py [hunk 2]
```diff
@@ -1,9 +1,53 @@
 import os
 import errno
+import time
+from functools import wraps
 
 from cortx.utils.conf_store import Conf
 from cortx.utils.conf_store.error import ConfError
 
+class ExponentialBackoff:
```
Comment (cdeshmukh) [inline py-utils/src/utils/common/common.py:9]: This is interesting and unique.
However latency for all these calls would change, these should be communicated to teams now

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**